cogs/fun.py

```diff
@@ -422,45 +424,102 @@ async def bill(self, ctx):
+			await ctx.send(embed=e)
+	
+	@commands.cooldown(rate=3, per=10.0, type=commands.BucketType.user)
+	@xkcd.command(aliases=['rand'])
+	async def random(self, ctx):
+		'''Returns a random xkcd comic'''
 		async with ctx.typing():
+			url = 'https://c.xkcd.com/random/comic/'
 			async with ctx.http.get(url) as resp:
 				if resp.status != 200:
 					raise commands.CommandError('Request failed.')
+				num = str(resp.url).lstrip('https://xkcd.com/').rstrip('/')
+			url = 'https://xkcd.com/{}/info.0.json'.format(num)
+			comic_json = await self.get_xkcd_json(ctx, url)
+			e = await self.make_xkcd_embed(comic_json)
+			await ctx.send(embed=e)
+	
+	@commands.cooldown(rate=1, per=7.0, type=commands.BucketType.user)
+	@xkcd.command(aliases=['search'])
+	async def relevant(self, ctx, *, search : str):
+		'''Grabs a relevant xkcd from `relevantxkcd.appspot.com`.'''
+		async with ctx.typing():
+			relevant_xkcd_url = 'https://relevantxkcd.appspot.com/process?action=xkcd&query='
+			search_url = relevant_xkcd_url + urllib.parse.quote(search)
+			async with ctx.http.get(search_url) as resp:
+				text = await resp.text()
+				results = text.split('\n')
+				num = results[2].split(' ')[0]
+				e = await self.get_xkcd_comic(ctx, num)
+				more_results = f'[More results]({search_url})'
+				relevance = '**Relevancy:** {}%\n{}\n\n'.format(str(round(float(results[0])*100,2)), more_results)
+				e.description = relevance + e.description
+				await ctx.send(embed=e)
+			
+	
+	async def get_xkcd_comic(self, ctx, id):
+		url = f'https://xkcd.com/{id}/info.0.json'
+		async with ctx.http.get(url) as resp:
+			if resp.status != 200:
+				raise commands.CommandError('Request failed.')
+			comic_json = await resp.json()
+		e = await self.make_xkcd_embed(comic_json)
+		return e
 
-				content = await resp.text()
-
-				comic_url = str(resp.url)
-
-			bs = BeautifulSoup(content, 'html.parser')
-			brs = bs.find('div', attrs=dict(id='middleContainer'))
-			img = brs.find('img')
-
-			if img is None:
-				await ctx.send(url)
-				return
 
-			e = discord.Embed(
-				title=img['alt'],
+	async def get_xkcd_json(self, ctx, url):
+		async with ctx.http.get(url) as resp:
+			if resp.status == 404:
+				raise commands.CommandError('Comic does not exist.')
+			if resp.status != 200:
+				raise commands.CommandError('Request failed.')
+			return await resp.json()
+	
+	async def make_xkcd_embed(self, comic_json):
+		comic_url = 'https://xkcd.com/{}'.format(comic_json['num'])
+		e = discord.Embed(
+				title=comic_json['title'],
 				url=comic_url,
-				description=img['title']
+				description='*{}*'.format(comic_json['alt'])
 			)
-
-			e.set_image(url='https:' + img['src'])
-			e.set_footer(text=comic_url.lstrip('https://').rstrip('/'), icon_url='https://i.imgur.com/onzWnfd.png')
-
-			await ctx.send(embed=e)
-
+		footer_text = 'xkcd.com/{}  •  {}-{}-{}'.format(comic_json['num'], comic_json['year'], comic_json['month'], comic_json['day'])
```

Comment:
Change this to use date so it is safely padded
```suggestion
		comic_date = date(int(comic_json['year']),int(comic_json['month']),int(comic_json['day']))
		footer_text = 'xkcd.com/{}  •  {}'.format(comic_json['num'], comic_date)
```